parlai/agents/seq2seq/seq2seq.py

```diff
@@ -37,70 +44,105 @@ def add_cmdline_args(argparser):
             help='which GPU device to use')
 
     def __init__(self, opt, shared=None):
+        # initialize defaults first
         super().__init__(opt, shared)
-        opt['cuda'] = not opt['no_cuda'] and torch.cuda.is_available()
-        if opt['cuda']:
-            print('[ Using CUDA ]')
-            torch.cuda.set_device(opt['gpu'])
         if not shared:
-            # don't enter this loop for shared (ie batch) instantiations
+            # this is not a shared instance of this class, so do full
+            # initialization. if shared is set, only set up shared members.
+            if opt.get('model_file') and os.path.isfile(opt['model_file']):
+                # load model parameters if available
+                print('Loading existing model params from ' + opt['model_file'])
+                new_opt, self.states = self.load(opt['model_file'])
+                # override options with stored ones
+                opt = self.override_opt(new_opt)
+
             self.dict = DictionaryAgent(opt)
             self.id = 'Seq2Seq'
-            hsz = opt['hiddensize']
+            # we use EOS markers to break input and output and end our output
```

Comment:
sure we could switch, the mcgill folks used eos and eou (end of utterance) in the ubuntu dataset, and douwe was saying we should have a "start of sentence" token as well, so I'm open to changing this to whatever is most useful / clear